ztp/gitops-subscriptions/argocd/Upgrade.md

```diff
@@ -1,43 +1,127 @@
+## Upgrading GitOps ZTP
+The Gitops ZTP infrastructure can be upgraded independently from the underlying cluster, ACM, and Openshift version running on the spoke clusters. This procedure guides the upgrade process to avoid impact on the spoke clusters, however any changes to the content or settings of policies, including adding recommended content will result in changes that must be rolled out and reconciled to the spokes.
 
-Follow the steps 1, 2 and 3 to prepare the hub cluster as [Readme](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/gitops-subscriptions/argocd/README.md) (Preparation of Hub cluster) section mentioned.
- 
-**Upgrading the SiteConfig repository:**
-In the siteConfig repository follow the steps below:
- 1. Delete the post-sync.yaml and pre-sync.yaml files.
- 1. If the siteconfig*.yaml in the repository contain definition for the Namespace remove it.
- 1. Update the kustomization.yaml file to include all SiteConfig yaml files in the `generators` section as below
+At a high level the strategy for upgrading the GitOps ZTP infrastructure is:
+1. Stop the ArgoCD applications
+1. Install the new tooling
+1. Update required content and optional changes in the GIT repository
+1. Update and restart the application configuration
+
+This procedure assumes you have a fully operational Hub cluster running the prior version of the GitOps ZTP infrastructure. If you need to install a new instance of the ZTP infrastructure, follow the instructions in the [Readme](README.md) section (Installing the GitOps Zero Touch Provisioning pipeline).
+
+### Upgrade Preparation
+Obtain the new version of the GitOps ZTP container. This container can be custom built using the ../../ztp/resource-generator build directory. Extract the argocd/deployment directory as described in [Preparation of ZTP GIT repository](#preparation-of-ztp-git-repository).
+
+Update the clusters-app.yaml and policies-app.yaml to reflect the name of your Applications and the URL, branch, and path for your GIT repository.
+
+### Remove obsolete content
+If the upgrade process results in obsolete policies, for example by moving policies from one namespace to another, the steps in this section should be taken prior to upgrade to remove those policies in an automated way. These steps need to be done prior to stopping the existing GitOps ZTP application (below) so the existing application can remove the affected resources.
+
+1. Remove the affected PolicyGenTemplate(s) from GIT, commit and push to the remote repository.
+1. Wait for the changes to synchronize through the application and the affected policies to be removed from the hub cluster.
+
+Note that removing these policies from GIT, and as a result also removing them from the hub cluster, will not affect any configuration of the managed spoke clusters. Removing a policy from the hub does not delete from the spoke cluster the CRs managed by that policy.
```

Comment:
In general, "Removing the policies from the hub doesn't remove them from the spoke" that might be not true. May be its better to phrase it like  "Note that removing the **ZTP DU profile policies** from GIT, and as a result also removing them ....."

Reply:
Done.